### PR TITLE
feat: search member within an organisation

### PR DESCRIPTION
**Pull request by olamstevy**
## What does the PR do?
This pull request adds a feature that allows users in an organisation search for other members in the organisation.

### How should this manually tested?
- A user must be logged in to access this endpoint.
- Logged in user must be in the say organisation as members that are being searched for.
- The logged in user makes a GET request to `/api/v1/organizations/{org_id}/members/search` containing the following payload:
**Query Parameter**: filter(optional) and can only be one of `active_member`, `suspended` or `left_workspace`. 
**Request Header** : Authorization `Bearer <token>`
**Request Body**
```JSON
{
    "search_term":  "name or email",
}
```

### Test Results with postman
**Searching for all member "user"**
Screenshot (28)

**Searching for suspended member**
Screenshot (31)

**Searching for active_members only**
Screenshot (32)

### Responses
**Successful Response**
```JSON
{
    "status": 200,
    "message": "User(s) found successfully",
    "data": {
        "members": [{
            "user_id": "some-user-uuid",
            "username": "username",
            "email": "user@example.org",
            "name": "firstname lastname",
            "phone_number": "23320320932",
            "profile_pic_url": "picture_url"
        }]       
    }
}
```

**Error (Bad Request)**
```JSON
{
    "status": 400,
    "message": "Bad Request"
}
```

**Error (Forbidden Request)**
```JSON
{
    "status": 403,
    "message": "User is not permitted to access this resource"
}
```

**Error (Authorization)**
```JSON
{
    "status": 401,
    "message": "User is currently unauthorized, kindly authenticate to continue"
}
```

Task

- [x] Implement an endpoint to allow users search for members in an organization. GET `/api/v1/organizations/{org_id}/members/search`
 - [x] Implement server-side logic to handle successful and failed get requests.
 - [x] Ensure that only logged in users can search for other members in the same organisations
 - [x] Write unit test to confirm accurate data is send when request is successful.
 - [x] Write unit test to verify appropriate status messages are contained in the response.

### CODE REVIEW

**Review by Heba-WebDev — APPROVED** [review]

File: src/modules/organisations/dto/search-member.dto.ts
Comment (CFCIfe) [inline src/modules/organisations/dto/search-member.dto.ts:1]: Can the DTOs be splitted

File: src/modules/organisations/organisations.controller.ts [hunk 1]
```diff
@@ -84,4 +85,21 @@ export class OrganisationsController {
...
     return this.organisationsService.getOrganisationMembers(org_id, page, page_size, sub);
   }
+
+  @Get(':org_id/members/search')
+  @ApiOperation({ summary: 'Search for members within an organization' })
+  @ApiResponse({ status: 200, description: 'User(s) found successfully', type: SearchMemberResponseDto })
+  @ApiResponse({ status: 400, description: 'Bad request' })
+  @ApiResponse({ status: 403, description: 'User does not have permission' })
+  @ApiResponse({ status: 500, description: 'An error occured internally' })
+  async searchMmebers(
```
Comment (CFCIfe) [inline src/modules/organisations/organisations.controller.ts:95]:
```suggestion
async searchMembers(
```

File: src/modules/organisations/organisations.controller.ts [hunk 2]
```diff
@@ -88,6 +92,22 @@ export class OrganisationsController {
     return this.organisationsService.getOrganisationMembers(org_id, page, page_size, sub);
   }
 
+  @UseGuards(MembershipGuard)
+  @Get(':org_id/members/search')
+  @ApiOperation({ summary: 'Search for members within an organization' })
+  @ApiResponse({ status: 200, description: 'User(s) found successfully', type: SearchMemberResponseDto })
+  @ApiResponse({ status: 403, description: 'User does not have permission' })
+  @ApiResponse({ status: 404, description: 'Organisation Not Found' })
+  @ApiResponse({ status: 500, description: 'An error occured internally' })
```
Comment (Cobidex) [inline src/modules/organisations/organisations.controller.ts:101]: Internal error isn't what you want your client to see in production. All endpoint is to expect this by default when the error isn't from the client. Please remove

File: src/modules/organisations/organisations.controller.ts [hunk 3]
```diff
@@ -88,6 +92,22 @@ export class OrganisationsController {
...
+  async searchMembers(
+    @Param('org_id') orgId: string,
```
Comment (Cobidex) [inline src/modules/organisations/organisations.controller.ts:103]: Please use validation pipes to ensure only valid UUID strings are allowed through. Else you'll raise exceptions from database validation.

File: src/modules/organisations/organisations.service.ts [hunk 4]
```diff
@@ -169,6 +171,43 @@ export class OrganisationsService {
     }
   }
 
+  async searchOrganisationMember(orgId: string, searchTerm: string, searchMemberQueryDto: SearchMemberQueryDto) {
+    const organisationMembers = await this.organisationMemberRepository.find({
```
Comment (Cobidex) [inline src/modules/organisations/organisations.service.ts:175]: All your logic should have ended here. There should only be one organisation member with a particular user and organisation id. Everything else after this should be you formating and returning your response. I think the rest of your code would be unnecessary if you simply optimize your query appropriately and use findOne instead

Comment (olamstevy) [reply src/modules/organisations/organisations.service.ts:178]: The purpose is to search organisation members not an organisation member

**Pull request closed without merging**